cluster-autoscaler/utils/drain/drain.go

```diff
@@ -113,19 +113,19 @@ func GetPodsForDeletionOnNodeDrain(
 
 		controllerRef := ControllerRef(pod)
 
-		if scaleDownNodesWithCustomControllerPods {
+		if skipNodesWithCustomControllerPods {
+			// TODO(vadasambar): remove this when we get rid of skipNodesWithCustomControllerPods
+			replicated, isDaemonSetPod, blockingPod, err = legacyCheckForReplicatedPods(listers, pod, minReplica)
+			if err != nil {
+				return []*apiv1.Pod{}, []*apiv1.Pod{}, blockingPod, err
+			}
+		} else {
 			if controllerRef != nil {
```

Comment:
Looks like `controllerRef` is not used anywhere else, so maybe just directly call `ControllerRef()` here?

Reply:
Updated. Thank you for pointing this out.